src/wiring/saveEntireWiring.ts

```diff
@@ -27,56 +31,204 @@ export default async function saveEntireWiring(
+    if (e.code !== "EEXIST") {
+      throw e;
+    }
+  }
+  const svgToImg = require("svg-to-img");
+  const jsdom = require("jsdom");
+  const { JSDOM } = jsdom;
+
   await writeFile(join(wiringPath, "toc.json"), JSON.stringify(toc, null, 2));
 
   for (let i = 0; i < toc.length; i++) {
     const doc = toc[i];
-    const sanitizedTitle = doc.Title.replace(/\//g, "-");
 
-    const newTitle = join(wiringPath, `${sanitizedTitle}.png`);
+    const sanitizedTitle = doc.Title.replace(/\//g, "-");
 
-    if (doc.Type === "Page") {
-      console.log(
-        `Downloading wiring manual page ${doc.Title} as image (#${doc.Number})`
+    // Connector views, different format handled later
+    if (doc.Type === "Page" && doc.Title !== "Connector Views") {
+      // Need pageList per docNumber
+      // Page lists for "Page" type documents is returned as ["001, "002", "003", etc]
+      const pageList = await fetchPageList(
+        {
+          book: fetchWiringParams.book,
+          cell: doc.Number,
+          title: doc.Title,
+          page: "1",
+          bookType: fetchWiringParams.bookType,
+          contentmarket: fetchWiringParams.contentmarket,
+          contentlanguage: fetchManualParams.contentlanguage,
+          languageCode: fetchWiringParams.languageCode
+        },
+        cookieString
       );
 
-      await fetchPage(
+      // Fetch each page per Wiring section
+      for (let j = 0; j < pageList.length; j++) {
+        const page = pageList[j].replace(/^0+/, "");
+        const sectionPath = join(wiringPath, sanitizedTitle);
+        try {
+          await mkdir(sectionPath);
+        } catch (e: any) {
+          if (e.code !== "EEXIST") {
+            throw e;
+          }
+        }
+        const newTitle = join(
+          sectionPath,
+          `${doc.Number}_${sanitizedTitle}_${page}.png`
+        );
+        console.log(
+          `Downloading wiring diagram Section: ${doc.Number} Title: ${doc.Title} Page: ${page} as image. `
+        );
+
+        const svgData = await fetchSvg(
+          doc.Number,
+          page,
+          fetchWiringParams.environment,
+          fetchManualParams.vehicleId,
+          fetchManualParams.WiringBookCode,
+          fetchWiringParams.languageCode,
+          cookieString,
+        );
+
+        // If the width/height are not specified when converting to png, the image ends up very small.
+        // Read the width/height from the viewbox of the svg so we can preserve the original size.
+        const svgParsed = new JSDOM(svgData, 'image/svg+xml').window.document;
+        const viewBox = svgParsed.getElementById("svgDoc").getAttribute("viewBox");
+        // viewbox is "x y width height"
+        const width = viewBox.split(' ')[2];
+        const height = viewBox.split(' ')[3];
+        const image = await svgToImg.from(svgData).toPng({
+          path: newTitle,
+          background: 'white',
+          height: height,
+          width: width
+        });
```

Comment:
I'm almost certain that we can do this inside Playwright: if we just call [page.setContent](https://playwright.dev/docs/api/class-page#page-set-content) on the SVG, that'll load it up in the browser. Then, we can query the browser for these SVG properties (if needed) and PDF it right from there. PDFing it also keeps them as vectors.